Semantic region renderer

Andreas-Forster opened the pull request:
Added a rendering option to render a specific texture image for all faces. This can be used to generate semantic region images. An example is the provided texture where the eyes, the lips, and the ears are colored differently than the skin.

Andreas-Forster committed:
render region maps in addition

Andreas-Forster committed:
removed bug

Andreas-Forster committed:
added missing data

Andreas-Forster committed:
corrected mapping .json

Andreas-Forster committed:
updated README

Andreas-Forster committed:
changed image size

BernhardEgger commented on data/config_files/example_config_controlled.json:
```diff
@@ -15,12 +15,17 @@
     "__comment-5" : "render-normals renders visualization of the vertex normals relative to the camera",
     "__comment-6" : "render-albedo renders the albedo of the face instance",
     "__comment-7" : "render-illumination renders a visualization of the illumination",    
+    "__comment-8" : "render-region-maps renders a visualization of the illumination",
```
renders a ...

BernhardEgger commented on data/config_files/example_config_random.json:
```diff
@@ -21,13 +21,15 @@
     "__comment-6" : "render-albedo renders the albedo of the face instance",
-    "__comment-7" : "render-illumination renders a visualization of the illumination",   
+    "__comment-7" : "render-illumination renders a visualization of the illumination",
+    "__comment-8" : "render-region-maps renders a visualization of the illumination",
```
renders a ...

BernhardEgger commented on data/config_files/example_config_controlled.json:
```diff
@@ -15,12 +15,17 @@
+    "render-illumination": true,
+    "render-region-maps": true
+  },
+  "correspondence-maps": {
```
is this needed?

BernhardEgger commented on src/main/scala/faces/utils/TextureMappedPropertyIO.scala:
```diff
@@ -0,0 +1,76 @@
+  }
+
+//
+//  val refWithTextureMapping = MeshIO.read(new File("/export/faces/model/model2012.2/reference/mean2012_l7_bfm_pascaltex.msh.gz")).get.texturedMesh3D.get
```
remove?

BernhardEgger commented on src/test/resources/controlledFacesSettings.json:
```diff
@@ -14,13 +14,15 @@
     "__comment-6" : "render-albedo renders the albedo of the face instance",
-    "__comment-7" : "render-illumination renders a visualization of the illumination",    
+    "__comment-7" : "render-illumination renders a visualization of the illumination",
+    "__comment-8" : "render-region-maps renders a visualization of the illumination",
```
renderas a...

BernhardEgger commented on src/test/resources/randomFacesSettings.json:
```diff
@@ -21,13 +21,15 @@
     "__comment-6" : "render-albedo renders the albedo of the face instance",
-    "__comment-7" : "render-illumination renders a visualization of the illumination",    
+    "__comment-7" : "render-illumination renders a visualization of the illumination",
+    "__comment-8" : "render-region-maps renders a visualization of the illumination",
```
renders a ...

BernhardEgger reviewed (commented):
I have some minor changes.
It would be very cool to get the bfm'09 face segments also as a default mask :)
https://faces.dmi.unibas.ch/bfm/main.php?nav=1-1-0&id=details

Currently the mask is rendered with shadowing and interpolation. For segmentation I think it might be useful to just get a pure color value without illumination effects (as done for other modalities). 
I'm not sure if there is an easy way remove interpolation between two regions

BernhardEgger commented:
Does the texture mapping work for both face 12 and bfm mask?

Andreas-Forster committed:
added bfm and face12 mappings and corrected things with regards to comments

Andreas-Forster commented:
All mentioned issues should be addressed:

- Shading removed from rendering.
- Added specific mappings for BFM and face12 masked models (adapted config files so the mappings match the model).
- Updated comments in config files.
- Removed unused section in a config file.
- Removed out-commented code.

Andreas-Forster committed:
added bfm09 region map

Andreas-Forster commented:
As an early Christmas present, there is now also the bfm09.png available as region map.

BernhardEgger committed:
Update README.md

BernhardEgger commented:
Thank you - now it looks great.
I changed a small typo in the readme.

Happy holidays and I like the present :)

AdamKortylewski reviewed (approved):
Thanks for the pull request, it's a great feature!
